src/main/kotlin/io/snyk/plugin/ui/jcef/ApplyFixHandler.kt

```diff
@@ -0,0 +1,215 @@
+
+            println("[applyFixHandler] Received request to apply fix on file: $filePath")
+            println("[applyFixHandler] Patch to apply: $patch")
+
+
+            // Avoid blocking the UI thread
+            CoroutineScope(Dispatchers.IO).launch {
+                try {
+                    val success = applyPatchAndSave(project, filePath, patch)
+
+                    val script = """
+                        window.receiveApplyFixResponse($success);
+                    """.trimIndent()
+
+                    withContext(Dispatchers.Main) {
+                        jbCefBrowser.cefBrowser.executeJavaScript(script, jbCefBrowser.cefBrowser.url, 0)
+                    }
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
+            }
+            return@addHandler JBCefJSQuery.Response("success")
+        }
+
+        return object : CefLoadHandlerAdapter() {
+            override fun onLoadEnd(browser: CefBrowser, frame: CefFrame, httpStatusCode: Int) {
+                if (frame.isMain) {
+                    val script = """
+                    (function() {
+                        if (window.applyFixQuery) {
+                            return;
+                        }
+                        window.applyFixQuery = function(value) { ${applyFixQuery.inject("value")} };
+                    })();
+                    """.trimIndent()
+                    browser.executeJavaScript(script, browser.url, 0)
+                }
+            }
+        }
+    }
+
+    private fun applyPatchAndSave(project: Project, filePath: String, patch: String): Boolean {
+        val virtualFile = findVirtualFile(filePath) ?: run {
+            println("[applyPatchAndSave] Virtual file not found for path: $filePath")
+            return false
+        }
+
+        println("[applyPatchAndSave] Found virtual file: $virtualFile")
+
+        val fileContent = ApplicationManager.getApplication().runReadAction<String?> {
+            val document = FileDocumentManager.getInstance().getDocument(virtualFile)
+            document?.text
+        } ?: run {
+            println("[applyPatchAndSave] Document not found or is null for virtual file: $filePath")
+            return false
+        }
+
+        println("[applyPatchAndSave] Initial file content: $fileContent")
+
+        val diffPatch = parseDiff(patch)
+        val patchedContent = applyPatch(fileContent, diffPatch)
+
+        println("[applyPatchAndSave] Patched content that will be written: $patchedContent")
+
+        // Apply the patch inside a WriteCommandAction
+        WriteCommandAction.runWriteCommandAction(project) {
+            val document = FileDocumentManager.getInstance().getDocument(virtualFile)
+            document?.let {
+                it.setText(patchedContent)
+
+                println("[applyPatchAndSave] Content after applying patch: ${it.text}")
+
+                FileDocumentManager.getInstance().saveDocument(it)
```

Comment:
We should not save the document after fixing. This can cause problems for user workflows.